src/components/BulkUploadListItem.tsx

```diff
@@ -0,0 +1,76 @@
+import React from 'react';
+import { CheckCircleIcon, XCircleIcon } from '@heroicons/react/24/outline';
+import { BulkUpload } from '../interfaces/BulkUpload';
+import './BulkUploadListItem.css';
+import { Spinner } from './Spinner';
+
+interface BulkUploadListItemProps {
+  upload: BulkUpload;
+}
+
+export const BulkUploadListItem = ({ upload }: BulkUploadListItemProps) => {
+  const {
+    fileName, fileSize, status, createdAt,
+  } = upload;
+
+  // TODO: Normalize these against the statuses returned by the API.
+  const isProcessing = status === 'processing';
+  const isComplete = status === 'complete';
+  const hasError = status === 'error';
+
+  const makeClassNames = () => {
+    const classNames = ['bulk-upload-list-item'];
+
+    classNames.push(status);
+
+    return classNames.join(' ');
+  };
+
+  const makeIcon = () => {
+    if (isProcessing) {
+      return <Spinner />;
+    }
+
+    if (isComplete) {
+      return <CheckCircleIcon className="icon" />;
+    }
+
+    if (hasError) {
+      return <XCircleIcon className="icon" />;
+    }
+
+    return null;
+  };
+
+  const makeMetadata = () => {
+    const metadata = [];
+
+    if (isProcessing) {
+      metadata.push(createdAt, fileSize, 'Processing…');
+    }
+
+    if (isComplete) {
+      metadata.push(createdAt, fileSize);
```

Comment:
@slifty this makes me think we want the `bulk_uploads` table to have a `finished_at` or so!

Reply:
Ah, that brings up a good note… I just YOLO'ed `createdAt` not knowing what the actual best timestamp attribute would be and meant to call attention to it. Thanks for noticing, and yes, any more appropriate stamp is fine.

I do think if we only show one timestamp, reporting to the user when _they_ took the action is preferable. But there are other good reasons for showing `finished_at`, including (probably) closer sync with the `created_at` timestamps of individual rows. In any case I doubt the differences will be material and am not at all insistent on one way or the other.

Reply:
Oh, wait. I just realized the "Failed" status is written as if it's reporting when the failure occurred. Hmm.

Okay I've thought myself into thinking that for pending/in_progress we should report the start time (or something close to upload time) and for the other statuses we should report the end time (or something close). I don't think this precision is vital to the UX, but if it's easy enough to accomplish, then it'd be good.

In any case, I don't want to the blocker on that, so go with whatever you and @slifty think makes sense!

Reply:
For the immediate term if we could work with just the `createdAt` value that is preferred, I agree we should aspire to include more information before we consider the feature complete.  (this also gives us enough time to think about whether it would be better to have that more robust "status reporting" / "job history" feature which would provide the information in a more flexible / granular way along with the other desired metadata)

Reply:
https://github.com/PhilanthropyDataCommons/service/issues/652